books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
+
+(If you're wondering why we don't just use `aspect_ratio` when computing `viewport_width`, it's
+because the _actual_ aspect ratio is the integer-based ratio of image width over image height, which
+will yield zero for values less than one.)
 
-I’ll put the “eye” (or camera center if you think of a camera) at $(0,0,0)$.
+Next we will define the camera center: a point in 3D space from which all scene rays will originate
+  (this is also commonly referred to as the _eye point_).
+The vector from the camera center to the viewport center will be orthogonal to the viewport.
+We'll initially set the distance between the viewport and the camera center point to be one unit.
+This distance is often referred to as the _focal length_.
+
+I’ll put the camera center at $(0,0,0)$.
 I will have the y-axis go up, and the x-axis to the right.
-In order to respect the convention of a right handed coordinate system, into the screen is the
-negative z-axis.
+Since we use a right-handed coordinate system, the vector into the screen is the negative z-axis.
+Note that I do not make the ray direction a unit length vector because I think not doing that makes
+  for simpler and slightly faster code.
+
+  ![Figure [camera-geom]: Camera geometry](../images/fig-1.03-cam-geom.jpg)
 
 Now the inevitable tricky part.
 While our 3D space has the conventions above, this conflicts with the standard _image_ coordinates,
-where the Y axis is inverted.
-In image coordinates, Y=0 denotes the top of the image, and Y=1 denotes the bottom.
-
-To help keep things straight, we'll compute two normalized (that is, in the range [0,1]) image
-coordinates: $s$ (horizontal) and $t$ (vertical).
-$s=0$ is image left, $s=1$ is image right, $t=0$ is image top, and $t=1$ is image bottom.
-That means that we get our vertical axis flip at the right point by computing $t = 1 - y$, where $y$
-is the vertical component of our 3D coordinate system, and $t$ is the flipped vertical component of
-our image coordinate system.
-
-I will traverse the screen from the upper left hand corner, and use two offset vectors along the
-screen sides to move the ray endpoint across the screen.
-Note that I do not make the ray direction a unit length vector because I think not doing that makes
-for simpler and slightly faster code.
+  where the Y axis is inverted (increasing going down).
+
+As we scan our image, we will start at the upper left pixel (pixel 0,0), scan left-to-right across
+  the row, and then scan the next row, top-to-bottom (reading order for most Latin languages).
+To help navigate the pixel grid, we'll use a vector from the left edge to the right edge
+  (`viewport_u`), and a vector from the upper edge to the lower edge (`viewport_v`).
 
-  ![Figure [cam-geom]: Camera geometry](../images/fig-1.03-cam-geom.jpg)
+Our pixel grid will be inset from the viewport edges by half the pixel-to-pixel distance, so each
+  pixel will gather light from the area around it.
```

Comment:
"Each Pixel will gather light from the area around it. If pixels were at the grid edge they would gather light from outside of the pixel grid. To avoid this, we will inset the pixels from the viewport edges by half the pixel-to-pixel distance"

Reply:
Alternively:

""Each Pixel will gather light from the area around it. If pixels were at the grid edge they would gather light from outside of the pixel grid. To avoid this, we will inset the pixels from the viewport edges"

Unless you acknowledge that the sampling area is a pixel in width, saying the inset length may not be needed (or helpful)

Reply:
There's nothing wrong with sampling light outside the pixel grid, or even the viewport. For example, the `sinc` image sampler (one of the best samplers available, though expensive) is infinite in extent. (See https://mathworld.wolfram.com/SincFunction.html.)

But yeah, could use another tweaking. How about this?
> Our pixel grid will be inset from the viewport edges by half the pixel-to-pixel distance.
> This way, our viewport area is evenly divided into width × height identical regions.